XamarinCommunityToolkit/Views/MediaElement/iOS/MediaElementRenderer.ios.cs

```diff
@@ -373,8 +384,34 @@ protected override void OnElementChanged(ElementChangedEventArgs<MediaElement> e
 
 				if (playedToEndObserver != null)
 				{
-					NSNotificationCenter.DefaultCenter.RemoveObserver(playedToEndObserver);
-					playedToEndObserver = null;
+					try
+					{
+						NSNotificationCenter.DefaultCenter.RemoveObserver(playedToEndObserver);
+						playedToEndObserver.Dispose();
+					}
+					catch
+					{
+					}
+					finally
+					{
+						playedToEndObserver = null;
+					}
+				}
+
+				if (rateObserver != null)
+				{
+					try
+					{
+						avPlayerViewController?.Player?.RemoveObserver(rateObserver, "rate");
+						rateObserver.Dispose();
+					}
+					catch
```

Comment:
What kind of exception do we want t cath here and in other places? Can we specify the type?
Also, I am not fancy of leaving **catch** blocks empty. Can we add a comment at least? "Suppress exception due to ...{reason}"

Reply:
Hi @AndreiMisiukevich, I just find we cannot catch any exceptions from there, so I remove all of them